theme/styles/_variables.sass

```diff
@@ -33,17 +36,21 @@ $c-text-light: #676767
 $c-text-back: #FFF
 
 // Breakpoints
-$break320: '(min-width: 320px)'
-$break480: '(min-width: 480px)'
-$break768: '(min-width: 768px)'
-$break992: '(min-width: 992px)'
-$break1200: '(min-width: 1200px)'
-
-$break320M: '(max-width: 319px)'
-$break480M: '(max-width: 479px)'
-$break768M: '(max-width: 767px)'
-$break992M: '(max-width: 991px)'
-$break1200M: '(max-width: 1199px)'
+$break320: 20em
+$break480: 30em
+$break768: 48em
+$break992: 62em
+$break1200: 75em
+$break1296: 81em
+$break1400: 87em
```

Comment:
em jednotky se chovají nepředvídatelně např. v Safari, viz https://adamwathan.me/dont-use-em-for-media-queries/ (+ https://bugs.webkit.org/show_bug.cgi?id=159635). Zároveň pokud by se změnila hodnota base font size, nebude sedět pojmenování breakpointu s jeho pixel hodnotou a bude to tak zavádějící.

Reply:
`$base-font-size` se nesmí měnit a používat jen minimálně. Věřím, že to není ani nikdy potřeba, kódu to neprospěje.

Root font size si vynucuje prohlížeč. Použití či změnění `$base-font-size` vede k nekonzistenci s nastavením prohlížeče. Pro běžné použití slouží `rem`.

`$base-font-size` se může hodit například pro utilitky jako `rem(...)`, pro převod pixelů z grafiky do relativních jednotek.

@JirkaVebr, @ViliamKopecky, @hexcross Jsme na stejné vlně?

Reply:
Článek jsem si znovu přečetl a stále se přikláním více k relativním jednotkám. Obzvlášť v době různých minidisplayů, výřezů, hustoty ~0.8 fyzických pixelů na jeden CSS pixel.

S `em` queries má problém pouze Safari, které je zoomnuté desktopovým stylem (~~pitch to zoom~~). Em queries se zoomem má nahlášené jako bug, takže ho v budoucnu může fixnout. Není to příliš velký edge case? Není místo toho lepší podporovat zařízení, která si třeba dnes neumíme ani představit nebo je prostě jen nevlastníme, a tak nejsme zvyklí, jak fungují? A co browserem vynucený font-size?

Reply:
> Není místo toho lepší podporovat zařízení, která si třeba dnes neumíme ani představit nebo je prostě jen nevlastníme, a tak nejsme zvyklí, jak fungují?

Předpoklad, že používáním `em` jednotek místo `px` v media queries tyto zařízení budeme podporovat, není platný. Už teď je jasná cesta, kterou tyto zařízení jdou (př. watchOS, ale v podstatě i iPhony), kdy si sami transformují hodnoty MQ a adaptují layout svým potřebám. Např. hodinky s watchOS 5 vygenerují stránku jako 320px širokou, ale pak jí zmenší na scale 0.49.

Ale ano, místo diskuze em vs. px je mnohem produktivnější věnovat lásku správnému postavení layoutu, který není ani tak závislý na MQ a umí si sám poradit s nedostatkem místa, rozmanitým obsahem a je přístupný.

Reply:
Taky myslím, že je lepší udělat v Safari trochu citlivější zoom než si znemožnit používat proměnné z kódu a odříznout i nezabugované prohlížeče s přednastavenou `$base-font-size`.

Reply:
> Z mého pohledu je mnohem pohodlnější, přehlednější používat em v media queries, když je používám i ve zbytku.

Což nemusí být pro všechny pravda. Já třeba používám px záměrně a toto přednastavení mi bude dělat nekonzistentní stav.

---

Nicméně, nikde tu nepadla výhoda mít MQ v em jednotkách, spíš jsme přišli na jeden zákeřný bug v Safari (BTW starý 2 roky). Když k tomu přidám nečitelnost této jednotky, nutnost ji překládat přes helper a hackování přechodné hodnoty pro max-width, jde to oproti metodice KISS a principům udržitelnému kódu obecně.

Reply:
> Což nemusí být pro všechny pravda. Já třeba používám px záměrně a toto přednastavení mi bude dělat nekonzistentní stav.

Ano. Už delší dobu se v media queries rozcházíme. Jako řešení vidím podporu v obou přístupech. Ve `variables` mít třeba jen `px` breakpointy nebo naopak jenom `em`, tak, aby se snadno dalo nahradit jedno za druhé. Případně duhou variantu nechat v komentářích.

Případně předpřipravené breakpointy z variables úplně zahodit. Ale čekám, že pro budu jen já. :(

-----------------------

1. `em` v media queries podporuje custom base font size vynucený prohlížečem/uživatelem.
2. Lépe se používá v kombinaci s `em` i ve zbytku stylů (`width`, `font-size`, ...). Konzistence.

```sass
.foo
   $break: 30em

   font-size: 1vw

   +media($break)
      font-size: $break
```

3. Bug v Safari je jen edge case. Je to malá oběť za body výše.

Reply:
Super, díky za příklady!

ad 1) a co když jako uživatel chci větší font ale ne větší UI? Protože bez toho funguje úplně stejně Page zoom funkce, která je dostupnější.

Klidně pojďme pevné breakpointy odstranit, správný přístup je přizpůsobovat MQ konkrétnímu obsahu.

Reply:
1) Podporovat větší font bez zvětšení UI je příliš složité. Velikost prvků musí často odpovídat velikosti font-size, počtu řádků. Uživatelů, kteří tohle chtějí je příliš málo. Ale máš pravdu, že snažit se dosáhnout stejného chování pro změnu font size jako při změně zoom je asi zbytečné. Beru první bod zpět. :)

Reply:
@JirkaVebr, @ViliamKopecky, @hexcross Jste někdo proti smazání předpřipravených breakpointů?

Reply:
@JirkaVebr Klidně předdefinované breakpointy media queries nechme. A které? `px`, `em` nebo oboje?